shm/inc/map.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "island.hpp"
+
+#include <vector>
```

Comment:
Recommended order is first header for this cpp,then standard library includes,then the rest files from project.

Reply:
Good recommendation!